addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3931,6 +3931,92 @@ options.registry.GalleryElement = options.Class.extend({
     },
 });
 
+options.registry.Button = options.Class.extend({
+    forceDuplicateButton: true,
+
+    /**
+     * @override
+     */
+    onBuilt(options) {
+        // Only if the button is built, not if a snippet containing that button
+        // is built (e.g. true if dropping a button from the snippet menu onto
+        // the page, false if dropping an "image-text" snippet).
+        if (options.isCurrent) {
+            this._adaptButtons();
+        }
+    },
+    /**
+     * @override
+     */
+    onClone(options) {
+        // Only if the button is cloned, not if a snippet containing that button
+        // is cloned.
+        if (options.isCurrent) {
+            this._adaptButtons();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Checks if there are buttons before or after the target element and
+     * applies appropriate styling.
+     *
+     * @private
+     */
+    _adaptButtons() {
+        const previousSiblingEl = this.$target[0].previousElementSibling;
+        const nextSiblingEl = this.$target[0].nextElementSibling;
+        let buttonNeighbor = false;
+        // When multiple buttons follow each other, they may break on 2 lines or
+        // more on mobile, so they need a margin-bottom. Also, if the button is
+        // dropped next to another button add a space between them.
+        if (previousSiblingEl?.matches(".btn")) {
+            previousSiblingEl.classList.add("mb-2");
+            buttonNeighbor = true;
+            this.$target[0].insertAdjacentHTML('beforebegin', '&nbsp;&nbsp;');
+        }
+        if (nextSiblingEl?.matches(".btn")) {
+            nextSiblingEl.classList.add("mb-2");
+            buttonNeighbor = true;
+            this.$target[0].insertAdjacentHTML('afterend', '&nbsp;&nbsp;');
+        }
+        if (buttonNeighbor) {
+            if (this.$target[0].matches(".s_custom_button")) {
+                this.$target[0].classList.remove(".s_custom_button");
+            } else {
+                // If the dropped button is not a custom button then we adjust
+                // its appearance to match its sibling.
+                [previousSiblingEl, nextSiblingEl].forEach((siblingEl) => {
```

Comment:
In fact, I don't understand this: to be changed after the merge please.

Why checking both neighbors? Should not it be `siblingEl = previousSiblingEl || nextSiblingEl` so we get only the previous one if it exists?